internal/istiooperator/merge_experimental.go

```diff
@@ -0,0 +1,117 @@
+//go:build experimental
+
+package istiooperator
+
+import (
+	"github.com/imdario/mergo"
+	operatorv1alpha2 "github.com/kyma-project/istio/operator/api/v1alpha2"
+	"github.com/kyma-project/istio/operator/internal/clusterconfig"
+	"istio.io/api/operator/v1alpha1"
+	istiov1alpha1 "istio.io/istio/operator/pkg/apis/istio/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"os"
+	"path"
+)
+
+func (m *IstioMerger) Merge(clusterSize clusterconfig.ClusterSize, istioCR *operatorv1alpha2.Istio, overrides clusterconfig.ClusterConfiguration) (string, error) {
+	toBeInstalledIop, err := m.GetIstioOperator(clusterSize)
+	if err != nil {
+		return "", err
+	}
+
+	ParseExperimentalFeatures(istioCR, &toBeInstalledIop)
+	mergedManifest, err := applyIstioCR(istioCR, toBeInstalledIop)
+	if err != nil {
+		return "", err
+	}
+	iopWithOverrides, err := clusterconfig.MergeOverrides(mergedManifest, overrides)
+	if err != nil {
+		return "", err
+	}
+	mergedIstioOperatorPath := path.Join(m.workingDir, MergedIstioOperatorFile)
+	err = os.WriteFile(mergedIstioOperatorPath, iopWithOverrides, 0o644)
+	if err != nil {
+		return "", err
+	}
+	return mergedIstioOperatorPath, nil
+}
+
+// ParseExperimentalFeatures parses experimental options defined in Istio CR
+// and sets the required features in the output operator CR.
+// Handles changes in ExperimentalFeaturesApplied condition which is only managed
+// in experimental flavour of image
+func ParseExperimentalFeatures(istioCR *operatorv1alpha2.Istio, op *istiov1alpha1.IstioOperator) {
+	if istioCR.Spec.Experimental == nil {
+		return
+	}
+	var errs []error
+	if istioCR.Spec.Experimental.EnableAlphaGatewayAPI {
+		err := enableGatewayAlphaAPI(op)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+	if istioCR.Spec.Experimental.EnableMultiNetworkDiscoverGatewayAPI {
+		err := enableMultiNetworkDiscoverGatewayAPI(op)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	// set condition based on errors collected
+	if istioCR.Status.Conditions == nil {
+		// lazy init conditions if nil
+		istioCR.Status.Conditions = new([]metav1.Condition)
+	}
+	if len(errs) > 0 {
+		cond := metav1.Condition{
+			Type:    ConditionExperimentalFeaturesApplied,
+			Status:  metav1.ConditionFalse,
+			Reason:  "ExperimentalFeaturesParseErrored",
```

Comment:
The reason confused me when I saw it in the test. Why do we call it parsing, when no parsing happens here? This might be confusing for a human consumer.